File: src/main/java/iot/tetracubered/iot/consumers/payloads/DeviceProvisioningPayload.java
package iot.tetracubered.iot.consumers.payloads;

import javax.json.bind.annotation.JsonbCreator;
import javax.json.bind.annotation.JsonbProperty;
import java.util.List;
import java.util.UUID;

public class DeviceProvisioningPayload {

    private final UUID id;
    private final String name;
    private final String feedbackTopic;
    private final List<DeviceFeatureProvisioningPayload> features;

    @JsonbCreator
    public DeviceProvisioningPayload(@JsonbProperty("id") UUID id,
                                     @JsonbProperty("name") String name,
                                     @JsonbProperty("feedback-topic") String feedbackTopic,
                                     @JsonbProperty("features") List<DeviceFeatureProvisioningPayload> features) {
        this.id = id;
        this.name = name;
        this.feedbackTopic = feedbackTopic;
        this.features = features;
    }

    public UUID getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getFeedbackTopic() {
        return feedbackTopic;
    }

    public List<DeviceFeatureProvisioningPayload> getFeatures() {
        return features;
    }
}

File: src/main/kotlin/red/tetracube/smarthub/iot/payloads/DeviceProvisioningPayload.kt
package iot.tetracubered.iot.payloads

import java.util.*
import javax.json.bind.annotation.JsonbCreator
import javax.json.bind.annotation.JsonbProperty

data class DeviceProvisioningPayload @JsonbCreator constructor(
    @field:JsonbProperty("id") val id: UUID,
    @field:JsonbProperty("name") val name: String,
    @field:JsonbProperty("feedback_topic") val feedbackTopic: String,
    @field:JsonbProperty("features") val features: List<DeviceFeatureProvisioningPayload>
)